plugins/module_utils/data_set.py

```diff
@@ -2092,6 +2093,119 @@ def _build_volume_string_iehprogm(volumes):
+            Name of a data set or alias.
+
+        Keyword Parameters
+        ------------------
+        tmp_hlq : str
+            Temp HLQ to use with mvscmdauth.
+
+        Returns
+        -------
+        tuple(bool, str)
+            A tuple containing whether name corresponds to a data
+            set alias and the name of the data set that the alias
+            points to.
+        """
+        # We need to unescape because this call to the system can handle
+        # special characters just fine.
+        name = name.upper().replace("\\", '')
+        idcams_cmd = f" LISTCAT ENTRIES('{name}') ALL"
+        response = DataSet._execute_idcams_cmd(idcams_cmd, tmp_hlq=tmp_hlq)
+
+        if response.rc > 0 or response.stderr_response != '':
+            raise MVSCmdExecError(
+                rc=response.rc,
+                stdout=response.stdout_response,
+                stderr=response.stderr_response
+            )
+
+        if re.search(r'(ALIAS -+)(1)', response.stdout_response):
+            base_name = re.search(
+                r'(ASSOCIATIONS\s*\n\s*[0-9a-zA-Z]+-+)([0-9a-zA-Z\.@\$#-]+)',
+                response.stdout_response
+            ).group(2)
+            return True, base_name
+        else:
+            return False, name
+
+    @staticmethod
+    def _execute_idcams_cmd(
+        cmd,
+        tmp_hlq=None,
+        space_primary=1,
+        space_type='k',
+        record_format='fb',
+        record_length=120
+    ):
+        """Runs an IDCAMS command using mvscmdauth's Python API.
+
+        Parameters
+        ----------
+            cmd : str
+                IDCAMS command to run.
+
+        Keyword Parameters
+        ------------------
+            tmp_hlq : str
+                Temp HLQ to use with mvscmdauth.
+            space_primary : int
+                Units of primary space for the input data set for IDCAMS.
+            space_type : str
+                Unit of data set space.
+            record_format : str
+                Record format for the input data set.
+            record_length : int
+                Record length for the input data set.
+
+        Returns
+        -------
+        ztypes.ZOAUResponse
+            Response object returned by mvscmd.execute_authorized.
+        """
+        temp_dd_location = None
+
+        try:
+            temp_dd_location = DataSet.create_temp(
+                hlq=tmp_hlq,
+                type='SEQ',
+                record_format=record_format,
+                space_primary=space_primary,
+                space_secondary=0,
+                space_type=space_type,
+                record_length=record_length
+            )
+
+            datasets.write(temp_dd_location, cmd)
+            cmd_dd = ztypes.DatasetDefinition(temp_dd_location, disposition='SHR')
+
+            dds = [
+                ztypes.DDStatement('SYSPRINT', '*'),
+                ztypes.DDStatement('SYSIN', cmd_dd)
+            ]
+
+            if tmp_hlq:
```

Comment:
Would be nice to have a comment linking back to the zoau issue but is no show stopper for me